client/verta/verta/_internal_utils/_utils.py

```diff
@@ -941,3 +941,34 @@ def is_org(workspace_name, conn):
     )
 
     return response.status_code != 404
+
+
+def as_list_of_str(tags):
+    """
+    Ensures that `tags` is a list of str.
+
+    Parameters
+    ----------
+    tags : str of list of str
+        If list of str, return unchanged. If str, return wrapped in a list.
+
+    Returns
+    -------
+    tags : list of str
+        Tags.
+
+    Raises
+    ------
+    TypeError
+        If `tags` is neither str nor list of str.
+
+    """
+    if isinstance(tags, six.string_types):
+        tags = [tags]
+    else:
```

Comment:
It might not be a list, right? For example, it could be a number or None, since some places are missing type checks.

Reply:
I intended for this to take any iterable of `str`, not just `list`s.
If it's, say, a number or `None`, then when the `for` loop begins, Python will raise
```
TypeError: 'int' object is not iterable
```

Two questions:

1. Is this adequate, or should I give the user a more specific error message, like "Expected `tags` to be a list of str, but got {}"?
1. Would be better to specifically only allow `list`s?

Reply:
1. As a user, I'd appreciate a more specific error message. If I get that one, I'll have to look into the client code to figure out what happened.
2. Lists and tuples, probably. Not sure, but seems like the base cases.

Reply:
I've decided to cast the arg to a `list` and catch the potential `TypeError`.

That way I can know what type is actually being returned from this fn, avoid any potential nasty interactions between arbitrary iterables and protobuf messages I pass them into, and avoid having to reset a generator after exhausting it in the `for` loop. Should be fine as long as users limit their tags to below a couple few hundred.

Reply:
Ugh no, that would accept `dict`s by only capturing its keys.

Reply:
`list`s, `tuple`s, `set`s. Final answer.